spicy/runtime/src/benchmarks/parsing.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2020-now by the Zeek Project. See LICENSE for details.
+
+#include <benchmark/benchmark.h>
+
+#include <hilti/rt/init.h>
+#include <hilti/rt/types/reference.h>
+#include <hilti/rt/types/stream.h>
+#include <hilti/rt/util.h>
+
+#include <spicy/rt/init.h>
+#include <spicy/rt/parsed-unit.h>
+#include <spicy/rt/parser.h>
+
+static std::string bigEndian(std::uint64_t number) {
+    char buffer[8];
+    for ( int i = 0; i < 8; ++i ) {
+        buffer[i] = static_cast<char>((number >> (56 - 8 * i)) & 0xFF);
+    }
+    return std::string(buffer, sizeof(buffer));
+}
+
+static std::string makeInput(std::uint64_t input_size) {
+    return hilti::rt::fmt("%s%sEND", bigEndian(input_size), std::string(input_size, 'A'));
+}
+
+template<class... Args>
+static void benchmarkParser(benchmark::State& state, Args&&... args) {
+    auto args_tuple = std::make_tuple(std::move(args)...);
+    auto parser_name = std::get<0>(args_tuple);
+
+    hilti::rt::init();
+    spicy::rt::init();
+
+    const spicy::rt::Parser* parser = nullptr;
+    for ( auto* p : spicy::rt::parsers() ) {
+        if ( p->name == parser_name ) {
+            parser = p;
+            break;
+        }
+    }
+
+    assert(parser);
```

Comment:
I'd do a hard error/abort here that triggers in release builds as well in case something goes wrong somewhere, just so that it's clear what happened.